core/modules/ccontrol/UserQuery.h

```diff
@@ -80,17 +80,31 @@ class UserQuery {
     // Delegate objects
     virtual std::shared_ptr<qdisp::MessageStore> getMessageStore() = 0;
 
+    /// This method should disappear when we start supporting results
+    /// in locations other than MySQL tables. We'll switch to getResultLocation()
+    /// at that point.
     /// @return Name of the result table for this query, can be empty
-    virtual std::string getResultTableName() = 0;
+    virtual std::string getResultTableName() const { return std::string(); }
+
+    /// Result location could be something like "table:table_name" or
+    /// "file:/path/to/file.csv".
+    /// @return Result location for this query, can be empty
+    virtual std::string getResultLocation() const { return std::string(); }
```

Comment:
This is not SELECT-specific, any query that returns data should provide location for its results. SHOW PROCESSLIST is one example of non-SELECT, there may be others in the future. `UserQuery` is only interface that we have today to provide all info to Czar, you cannot remove methods from this interface without breaking czar.